packages/mobile/src/firebase/saga.ts

```diff
@@ -158,9 +164,25 @@ export function* watchLanguage() {
   yield takeEvery(AppActions.SET_LANGUAGE, syncLanguageSelection)
 }
 
+export function* startFirebaseOnRefresh(channel: EventChannel<Notification>) {
+  while (true) {
+    const { data } = yield take(channel)
+    if (!data) {
+      break
+    }
+    Logger.info(`${TAG}/startFirebaseOnRefresh`, 'Notification received in the channel')
+    yield handleNotification(data.notification, NotificationReceiveState.APP_ALREADY_OPEN)
+  }
+}
+
+function* watchStartFirebaseOnRefresh() {
+  yield takeEvery(Actions.START_FIREBASE_ON_REFRESH, startFirebaseOnRefresh)
```

Comment:
Do we need a watcher here? Seems like we should just start the channel listener once when firebase is initialized.